mcon/parse.c

```diff
@@ -285,6 +285,54 @@ static int mongo_process_option(mongo_con_manager *manager, mongo_servers *serve
 }
 
 /* Option parser helpers */
+static int parse_auth_mechanism_properties(mongo_con_manager *manager, mongo_servers *servers, char *value, char **error_message)
+{
+	char *start, *end, *colon, *prop_name, *prop_value;
+
+	/* ignore empty value */
+	if (strlen(value) == 0) {
+		return 0;
+	}
+
+	start = value;
+
+	do {
+		end = strchr(start, ',');
+		colon = strchr(start, ':');
+
+		if (!colon) {
+			int len = strlen(start) + sizeof("Error while trying to parse auth mechanism properties: No separator for ''");
+
+			*error_message = malloc(len + 1);
+			snprintf(*error_message, len, "Error while trying to parse auth mechanism properties: No separator for '%s'", start);
```

Comment:
I also found it odd that we `malloc()` to the exact length here, but elsewhere (e.g. for unsupported options) use a constant buffer of 256 and allow the error message to be truncated.

In any event, I'm not sure a constant for the error message would help unless we want to define it with `"%s"` included and subtract 2 when calculating the length.